grails-app/controllers/org/pih/warehouse/user/UserController.groovy

```diff
@@ -266,32 +267,31 @@ class UserController {
 
 
     def disableLocalizationMode = {
+        // If we are in localization mode and want to change the language through the footer on React side
+        // we want to make sure we set the language to the clicked one, not for the previous locale
+        // we want to set the prev locale only if we disable localization mode through "Disable localization mode" button
+        Locale localeToSet = params?.localeToSet ? new Locale(params.localeToSet) : session?.prevLocale
+        session.locale = localeToSet
         session.useDebugLocale = false
-        if (!grailsApplication.config.openboxes.locale.custom.enabled) {
-            flash.message = "${g.message(code: 'localization.invalid.custom.message')}"
-        }
-
         def referer = request.getHeader("Referer")
         if (referer) {
             redirect(url: referer)
-        } else {
-            redirect(controller: "dashboard", action: "index")
+            return
         }
+        redirect(controller: "dashboard", action: "index")
     }
 
     def enableLocalizationMode = {
+        // We want to store the previous locale, so we can go back to it when disabling localization mode through button
+        session.prevLocale = localizationService.getCurrentLocale()
```

Comment:
Sorry to be nit-picky but let's avoid using abbreviations. `session.previousLocale` is more readable.

Reply:
Also is there any difference between getting the current locale versus using `session.locale`?

Reply:
As for this case probably not, because to be in `disableLocalizationMode` we have to be first in `enableLocalizationMode`, and in `enableLocalizationMode` we set the `session.locale`.
It was just a bit of overthinking by me probably, as I wanted to be "safe" that we might not have `session.locale` yet, because when you log in and do not neither change the language nor enable localization mode, your locale is taken from `session.user.locale` as `session.locale` does not yet exist.